### PR TITLE
Lesson 6 edmond sabou

### PR DESCRIPTION
**Pull request by edisabou**

### CODE REVIEW

File: lesson_6/edmond_sabou/functions2.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    ('XK', ['0', '0', '0', '0', '0', '0', '89', '93', '93']),
+]
+
+
+#######################################################################################################################
+def create():
+    new_dict = {country: data_list for country, data_list in raw_data}
+
+    return {
+        country: [
```
Comment (Liitle) [inline lesson_6/edmond_sabou/functions2.py:49]: you can also generate the dictionaries from the iteration. it is not necessary to manually increment de index of data_list

```
{ country:  [
    {"year":str(year), "coverage": data}
    for year, data in enumerate(data_list,start=2011)
  ] for country, data_list in new_dict.items()}
```
try to run this and see the result :D

File: lesson_6/edmond_sabou/functions2.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# >>> {'2019': [('Romania', 84), ('Germany', 95), ..., ('Latvia', 85)]}
+#
+def get_year_data(new_data_set, year):
+    #    print(type(new_data_set))
+    #    print(new_data_set)
+    new = {
+        year: [(country, index["coverage"])
+               for country in new_data_set.keys()
+               for index in new_data_set[country]
+               if index["year"] == str(year)]  # Q: De ce este nevoie sa pun str(year)??
```
Comment (Liitle) [inline lesson_6/edmond_sabou/functions2.py]: if you call the function get_year_data with year as string you don't have to convert it to string

File: lesson_6/edmond_sabou/functions2.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    new2 = {
+        country: [(value["year"], value["coverage"])
+                  for value in new_data_set[country]]
+    }
+
+    return new2
+
+
+#######################################################################################################################
+def perform_average(country_data):  # e urat.. stiu, non-pythonic, dar numai asa am reusit
```
Comment (Liitle) [inline lesson_6/edmond_sabou/functions2.py]: the logic is good, try to leave the comments in English 😄  and it is preferred to have the comment above the function definition

**Review by Liitle — CHANGES_REQUESTED** [review]
Nice work. I like that you improved the assignment. If you make the change to create function this is ready for merge

File: lesson_6/edmond_sabou/functions2.py (hunk 1 above)
Comment (edisabou) [reply lesson_6/edmond_sabou/functions2.py:49]: yes, I get the same result and it looks way more elegant. I've implemented this change.

File: lesson_6/edmond_sabou/functions2.py (hunk 3 above)
Comment (edisabou) [reply lesson_6/edmond_sabou/functions2.py]: understood

**Review by mariuspodean — APPROVED** [review]

**Pull request merged**